docs/features/message-pumps/service-bus.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "Azure Service Bus Message Pump"
+layout: default
+---
+
+# Azure Service Bus Message Pump
+
+Azure Service Bus Message Pump will perform all the plumbing that is required for processing queues & topics:
+
+- Manage message pump lifecycle
+- Deserialize messages into concrete types
+- Interpret message to provide correlation & context information
+- Provide exception handling
+- Provide telemetry
+
+As a user, the only thing you have to do is **focus on processing messages, not how to get them**.
+
+You can do this by creating a message handler which derives from `AzureServiceBusMessagePump<TMessage>`
+
+Here is an example of a message handler that expects messages of type `Order`:
+
+```csharp
+public class OrdersMessageHandler : AzureServiceBusMessagePump<Order>
+{
+    public OrdersMessageHandler(IConfiguration configuration, IServiceProvider serviceProvider, ILogger<OrdersMessageHandler> logger)
+        : base(configuration, serviceProvider, logger)
+    {
+    }
+
+    protected override async Task ProcessMessageAsync(Order orderMessage, AzureServiceBusMessageContext messageContext, MessageCorrelationInfo correlationInfo, CancellationToken cancellationToken)
+    {
+        Logger.LogInformation("Processing order {OrderId} for {OrderAmount} units of {OrderArticle} bought by {CustomerFirstName} {CustomerLastName}", orderMessage.Id, orderMessage.Amount, orderMessage.ArticleNumber, orderMessage.Customer.FirstName, orderMessage.Customer.LastName);
+
+        // Custom logic
+
+        Logger.LogInformation("Order {OrderId} processed", orderMessage.Id);
+    }
+}
+```
+
+As of today, the message handler is tightly coupled to the broker but overtime they will be decoupled.
+
+Over time you'll be able to:
+- Bring your own deserialization
+- Re-use a message handler across different brokers
+- Use multiple message handlers and let the pump route the messages to the correct handler.
+This can be based on message type, message context or custom message flow determination
+
+## Configuration
+
+Once the message handler is created, you can very easily configure it:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // Add Service Bus Queue message pump and use OrdersMessageHandler to process the messages
+    // ISecretProvider will be used to lookup the connection string scoped to the queue for secret ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING
+    services.AddServiceBusQueueMessagePump<OrdersMessageHandler>("ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
```

Comment:
The syntax here will change of course; but it's OK to have it like this for now.

Reply:
Agreed, just adding this to have something for now!